### ISSUE TITLE
I saw that your app was removed from Google Play. What is the reason?

### ISSUE THREAD

**Issue by NewMacPro** [alexch33/super-video-downloader#5]
I saw that your app was removed from Google Play. What is the reason?
Thanks

**Comment by alexch33** [comment]
Reason is new google play policy about revealing personal address to the whole internet, it's human's right violation, i've removed all my apps from play market

**Comment by NewMacPro** [comment]
> Reason is new google play policy about revealing personal address to the whole internet, it's human's right violation, i've removed all my apps from play market

Is it because Google Play Developer has reached the deadline for verifying personal information?
Then what is the status of your current developer account?

**Comment by alexch33** [comment]
> > Reason is new google play policy about revealing personal address to the whole internet, it's human's right violation, i've removed all my apps from play market
> 
> Is it because Google Play Developer has reached the deadline for verifying personal information? Then what is the status of your current developer account?

no, i removed apps by my own hands, my developer account status is good and verified, after verification i'd saw that my personal address available to whole internet and it is according google rules..., after that i removed all apps, this is riot against stupid rules, i want work with google anymore

**Comment by alexch33** [comment]
Also, for all who want to implement app in such category, like video downloader, for play market. There is huge amount of bots, that installing and removing app every day, trying crush app's indexing on early stages, for protect from such things you should have core auditory before launch, this crucial thing for success in my opinion

**Comment by ali-shakeel880** [comment]
I am in the process of developing an application that utilizes the YouTube-DL tool. However, the app includes a strict check to ensure it does not allow the downloading of videos from YouTube or violate any of their terms of service. Before proceeding with submission to the Play Store, I would like to clarify if the use of YouTube-DL in this capacity could potentially pose any risks to my developer account, even though the app will not permit the downloading of YouTube content.

Could you please provide guidance on whether this approach complies with Google Play Store policies, and if there are any potential risks to my Google Play Console account?

Thank you for your assistance.

**Comment by ali-shakeel880** [comment]
Can you provide me your email or any Contact details so that we can have  a meeting regarding to this issue.

**Comment by alexch33** [comment]
@ali-shakeel880  When my app was on market i had code like ```if (host == "youtube") { throw Exeption("Youtube not allowed") }``` I forgot several times and my app's updates was rejected, i thought this normal, i had no worries. Rule is simple, you right, youtube not allowed, this all i know, i think, if you app became popular...,  i dont know

**Comment by alexch33** [comment]
if you mean using library youtube-DL especially is allowed in "terms and conditions", for example youtube not allowed, potentially many others could not allowed, but no checks for this. Any others services constantly improving their systems and you must always do upgrades, for example right now not working anime services from screenshots, and i too busy, to investigate what they did,  and calm, because app not on market. As resume i think this apps category is grey zone and now best idea for making money on market, this my opinion

**Comment by alexch33** [comment]
if you mean using library youtube-DL especially is allowed in "terms and conditions", for example youtube not allowed, potentially many others could not allowed, but no checks for this. Any others services constantly improving their systems and you must always do upgrades, for example right now not working anime services from screenshots, and i too busy, to investigate what they did,  and calm, because app not on market. As resume i think this apps category is grey zone and now best idea for making money on market, this my opinion

**Comment by ali-shakeel880** [comment]
> @ali-shakeel880 When my app was on market i had code like `if (host == "youtube") { throw Exeption("Youtube not allowed") }` I forgot several times and my app's updates was rejected, i thought this normal, i had no worries. Rule is simple, you right, youtube not allowed, this all i know, i think, if your app became popular..., i dont know

Thank you for your assistance. I am currently implementing to display a dialog when the link is from YouTube, indicating that downloading videos from YouTube is not permitted. Can you provide me any other suggestion?
.

**Comment by ali-shakeel880** [comment]
> if you mean using library youtube-DL especially is allowed in "terms and conditions", for example youtube not allowed, potentially many others could not allowed, but no checks for this. Any others services constantly improving their systems and you must always do upgrades, for example right now not working anime services from screenshots, and i too busy, to investigate what they did, and calm, because app not on market. As resume i think this apps category is grey zone and now best idea for making money on market, this my opinion

Thank you for sharing your insights! I completely understand the challenges you’ve mentioned regarding the use of libraries like youtube-dl and the constant need for updates in response to service changes. It’s true that navigating the grey areas of app development can be tricky, but it also presents unique opportunities for innovation. I believe that with careful consideration and proactive management, we can create a compelling app that resonates with users and stands out in the market. But many apps are using this library i think and have 100 million plus downloads they are using it in web view and user do not need to login their accounts for video downloading like (instagram ,facebook ) accounts .My main focus is on that the user do not have to login their accounts to download a video they instead just paste the link and download the video ! i have seen many apps doing the same but I searched the whole google but i do not find anything that targets most of the platforms except the youtube-dl library.I thought that the others apps are using that library also they are also checking for the you-tube video and prompting user that youtbe vdeio cannot be downlaod . what are your suggestions that they are also using the youtube-dl? i am giving a reference app link **https://play.google.com/store/apps/details?id=video.downloader.videodownloader&pcampaignid=web_share** .It will be great help for me If you tell me that are they using the youtube-dl library.

**Comment by alexch33** [comment]
> > @ali-shakeel880 When my app was on market i had code like `if (host == "youtube") { throw Exeption("Youtube not allowed") }` I forgot several times and my app's updates was rejected, i thought this normal, i had no worries. Rule is simple, you right, youtube not allowed, this all i know, i think, if your app became popular..., i dont know
> 
> Thank you for your assistance. I am currently implementing to display a dialog when the link is from YouTube, indicating that downloading videos from YouTube is not permitted. Can you provide me any other suggestion? .

Always glad to help, i think this is good idea, i have no any suggestions, i tell you more, few times, before this was automated in play market, my app was in production with working youtube and when i upload update also with youtube, it was detected and only update was rejected, but app still was in production available.

P.S. Once i created youtube video with process of anime downloading and uploaded to youtube and got strike very fast :-)

**Comment by alexch33** [comment]
> > if you mean using library youtube-DL especially is allowed in "terms and conditions", for example youtube not allowed, potentially many others could not allowed, but no checks for this. Any others services constantly improving their systems and you must always do upgrades, for example right now not working anime services from screenshots, and i too busy, to investigate what they did, and calm, because app not on market. As resume i think this apps category is grey zone and now best idea for making money on market, this my opinion
> 
> Thank you for sharing your insights! I completely understand the challenges you’ve mentioned regarding the use of libraries like youtube-dl and the constant need for updates in response to service changes. It’s true that navigating the grey areas of app development can be tricky, but it also presents unique opportunities for innovation. I believe that with careful consideration and proactive management, we can create a compelling app that resonates with users and stands out in the market. But many apps are using this library i think and have 100 million plus downloads they are using it in web view and user do not need to login their accounts for video downloading like (instagram ,facebook ) accounts .My main focus is on that the user do not have to login their accounts to download a video they instead just paste the link and download the video ! i have seen many apps doing the same but I searched the whole google but i do not find anything that targets most of the platforms except the youtube-dl library.I thought that the others apps are using that library also they are also checking for the you-tube video and prompting user that youtbe vdeio cannot be downlaod . what are your suggestions that they are also using the youtube-dl? i am giving a reference app link **https://play.google.com/store/apps/details?id=video.downloader.videodownloader&pcampaignid=web_share** .It will be great help for me If you tell me that are they using the youtube-dl library.

O, finally i am understand meaning of downloader apps with link only, i wish you good luck!!! as i know all using youtube-dl or youtube-dlp, but on backend, my vision of such apps is different, having ability to download from private social media groups or premium content or content when you need to login should be but this is impossible for apps with download link only, for example you cant download 22 episode of anime X in player Y, if link same for different episodes

**Comment by ali-shakeel880** [comment]
I understand but I’m not considering all the video platforms i’m just
targeting few like insta facebook dailymotion snack video vimeo and other
such type of platforms only They all have unique id in theri links.

**Comment by ali-shakeel880** [comment]
Thanks for clearing that most of the  downloading apps are using youtube-dl
for downloading videos
But they are using this library with consequences like (for example video
cannot be downloaded from youtube) . I will take care for this type of
platform using this library.

**Comment by astalkb** [comment]
Can you please fix the auto screen rotation.

**Comment by alexch33** [comment]
> Can you please fix the auto screen rotation.

Hi! yes, i can, but little bit latter :)

**Comment by alexch33** [comment]
> Can you please fix the auto screen rotation.

I have added `lock screen in portrait mode` setting to settings, by default this turned off, so you should turn on

**Comment by DavidAriyan** [comment]
Off course, Goggle Play always removes actually good app and games. And left many fake apps untouched. It's crazy. That's why I get my favorite apps and games from outside playsl store.

**Comment by shibaFoss** [comment]
I believe the key here is to avoid allowing YouTube downloading at all costs. We could use the library, but only for safe sites. If we do this, Google shouldn't raise any issues. Am I right?

**Comment by alexch33** [comment]
> I believe the key here is to avoid allowing YouTube downloading at all costs. We could use the library, but only for safe sites. If we do this, Google shouldn't raise any issues. Am I right?

Based on my experience, looks like yes, you are, there is no another info

**Comment by woheller69** [comment]
what about publishing it on F-Droid instead?

**Comment by alexch33** [comment]
> what about publishing it on F-Droid instead?

Yes, i am planning to publish it on F-droid, but little bit latter, some configs must be implemented

**Comment by alexch33** [comment]
> what about publishing it on F-Droid instead?

Now it is available on [F-Droid](https://f-droid.org/packages/com.myAllVideoBrowser)

**Comment by Aqua0410** [comment]
> [@ali-shakeel880](https://github.com/ali-shakeel880) When my app was on market i had code like `if (host == "youtube") { throw Exeption("Youtube not allowed") }` I forgot several times and my app's updates was rejected, i thought this normal, i had no worries. Rule is simple, you right, youtube not allowed, this all i know, i think, if your app became popular..., i dont know

where to apply this?

**Comment by shibaFoss** [comment]
I also built a simple, privacy-friendly video downloader app for Android. It was on the Play Store for a while but got removed due to their content policies. 😔 So you're planning, be prepared. 

Now I made it open-source! and it supports all major sites, including YouTube, Instagram, Facebook, etc. Check it out here:  [https://github.com/shibaFoss/VideoMate](https://github.com/shibaFoss/VideoMate)

**Comment by rasool1947** [comment]
i think current fb downloader not working..


i am stuck how the in browser we detect video url of current focused video..
  the rest work is easy just to proivde it ytdl..
but link detection is challeng

**Comment by alexch33** [comment]
> i think current fb downloader not working..
> 
> i am stuck how the in browser we detect video url of current focused video.. the rest work is easy just to proivde it ytdl.. but link detection is challeng

Hi! facebook video detection working ONLY using youtube-dlp(i had abandon manual video detection for popular web-sites very long time ago, because they constantly upgrading breaks detection) 

But. I can't reproduce this, because for me facebook downloading working fine

<img width="1440" height="3120" alt="Image" src="https://github.com/user-attachments/assets/aec439cb-6faf-4615-82ae-e883bf2bbb52" />

**Comment by alexch33** [comment]
> I also built a simple, privacy-friendly video downloader app for Android. It was on the Play Store for a while but got removed due to their content policies. 😔 So you're planning, be prepared.
> 
> Now I made it open-source! and it supports all major sites, including YouTube, Instagram, Facebook, etc. Check it out here: https://github.com/shibaFoss/VideoMate

interesting, what polices are? i was on market almost year and nothing

**Comment by rasool1947** [comment]
will you guid me ..in browser how we can detetect video url...
once url detected then its easy to provide to ytdl ..

**Comment by alexch33** [comment]
> will you guid me ..in browser how we can detetect video url... once url detected then its easy to provide to ytdl ..

just go to web page with video and it should be detected, like on screenshot

<img width="1440" height="3120" alt="Image" src="https://github.com/user-attachments/assets/6b6c5cae-ed2a-4b55-82c3-3bd4e654dd79" />

and check that youtube-dlp turned on in settings(video detection by url)

**Comment by rasool1947** [comment]
ahaha not like that .
Sir in coding..
     just a todo note to mention here.. i will code by myslef..
just guide me step vice..

not asking full code ..just steps to follow

Thanks

**Comment by alexch33** [comment]
> ahaha not like that . Sir in coding.. just a todo note to mention here.. i will code by myslef.. just guide me step vice..
> 
> not asking full code ..just steps to follow
> 
> Thanks

ahah, understood;-)

idea behind video url detection is pretty simple: in method `shouldIntreceptRequest` check every request. If request urls ends with "m3u8" or "mpd" OR has m3u8 or mpd content inside -> send to youtube-dlp(cookies should be handled separately)

if request url has Content-Type is Video -> send it to regular downloader, with cookies and headers

I am not sure, that i have right understanding what are you asking, feel free ask additional questions if needed, i glad to help

**Comment by alexch33** [comment]
if you asking about facebook video detection without youtube-dlp: Facebook has huge json in page sources, this should be parsed and url extracted, but this very annoying task, may some LLM will help

**Comment by rasool1947** [comment]
Sir i am stuck at point ..in fb
      when i open fb and play video A from feed page in full screen it doanload Video A..
now i scrol to next video b in full screen page (player screen)  its not update the url to video B..still it showing video A link..

a tried with AI but they also did not solve it..

how other does this.. linke inshot app..

anything yoy know about this issues..

**Comment by alexch33** [comment]
> Sir i am stuck at point ..in fb when i open fb and play video A from feed page in full screen it doanload Video A.. now i scrol to next video b in full screen page (player screen) its not update the url to video B..still it showing video A link..
> 
> a tried with AI but they also did not solve it..
> 
> how other does this.. linke inshot app..
> 
> anything yoy know about this issues..

on FB using urls intercepting you don't achieve your goals. I know nothing about your are talking about, but facebook working using youtube-dlp, when page changed and page's url updated, youtube-dlp checking on video new url, but if video changed and web page's didnt update it will not send to youtube-dlp on check.

so, you could try to listen changes using JS interface of webview and try to intercept new url from java script

**Comment by alexch33** [comment]
My efforts to draw attention to human rights violations have led to nothing for more than a year and a half. I think there's no point in continuing. Moreover, Google has been a great help to me lately; I might even return the app to the Market. so, i am closing this issue
